pyqg/model.py

```diff
@@ -168,25 +172,44 @@ def __init__(
         ntd : int
             Number of threads to use. Should not exceed the number of cores on
             your machine.
-        q_parameterization : function
-            Optional function which takes the model as input and returns a `numpy`
-            array of shape (`nz`, `ny`, `nx`) to be added to dq/dt before stepping
-            forward. This can be used to implement subgrid forcing
+        q_parameterization : function or pyqg.Parameterization
+            Optional :code:`Parameterization` object or function which takes
+            the model as input and returns a :code:`numpy` array of shape
+            :code:`(nz, ny, nx)` to be added to :math:`\partial_t q` before
+            stepping forward.  This can be used to implement subgrid forcing
             parameterizations.
-        uv_parameterization : function
-            Optional function which takes the model as input and returns a tuple of
-            two `numpy` arrays, each of shape (`nz`, `ny`, `nx`), to be added to
-            the zonal and meridional velocity derivatives (respectively) at each
-            timestep. This can also be used to implemented subgrid forcing
-            parameterizations, but expressed in terms of velocity rather than
-            potential vorticity.
+        uv_parameterization : function or pyqg.Parameterization
+            Optional :code:`Parameterization` object or function which takes
+            the model as input and returns a tuple of two :code:`numpy` arrays,
+            each of shape  :code:`(nz, ny, nx)`, to be added to the zonal and
+            meridional velocity derivatives (respectively) at each timestep (by
+            adding their curl to :math:`\partial_t q`).  This can also be used
+            to implemented subgrid forcing parameterizations, but expressed in
+            terms of velocity rather than potential vorticity.
+        parameterization : pyqg.Parameterization
+            An explicit :code:`Parameterization` object representing either a
+            velocity or potential vorticity parameterization, whose type will
+            be inferred.
         """
 
         if ny is None:
             ny = nx
         if W is None:
             W = L
 
+        # if an explicit parameterization object was passed without a given
+        # type, infer it from its attributes
+        if parameterization is not None:
+            ptype = parameterization.parameterization_type 
```

Comment:
I think this line will error if you just pass a regular function. better to use
```suggestion
            ptype = getattr(parameterization, "parameterization_type", None)
```